docs: add Git branching best practices and optimize Netlify deployment instructions (#29)
CodeYourFuture/Project-TV-Show: levels/level-500.md

Here is one example layout.

![Screenshot of a website with a drop-down list with the show "Breaking Bad" selected](example-screenshots/example-level-500.jpg)

---

## Final Step: Professional Deployment to Netlify

> [!IMPORTANT]  

> > **Only proceed with this step once all the requirements above are fully functional and tested on your GitHub Pages site.** > This is your final "production" deployment.

Reviewer: ```suggestion
> **Only proceed with this step once all the requirements above are fully functional and tested on your GitHub Pages site.**
> This is your final "production" deployment.
```

> To complete the project, we will move from GitHub Pages to **Netlify** to practice using a professional hosting platform.

Reviewer: ```suggestion
To complete the project, we will move from GitHub Pages to **Netlify** to practice using a more advanced hosting platform.
```

### How to deploy to Netlify:

1. **Log in:** Go to [app.netlify.com](https://app.netlify.com/) and log in with your GitHub account.
2. **Add a new site:** Click on **"Add new site"** and select **"Import an existing project"**.
3. **Connect GitHub:** Choose GitHub as your provider and select your `Project-TV-Show` repository.
4. **Site Settings:** - Ensure the branch to deploy is `main`.
   - Leave "Build command" and "Publish directory" empty.
   - Click **"Deploy Project"**.
5. **Customize URL:** - Go to **Site configuration** > **Site details** > **Change site name**.
   - Change it to: `cyf-USERNAME-tv.netlify.app` (replacing `USERNAME` with your GitHub username).

> For more details, refer to the [Official CYF Netlify Guide](https://curriculum.codeyourfuture.io/guides/deployment-netlify/).

Reviewer: ```suggestion
Follow the [Official CYF Netlify Deployment Guide](https://curriculum.codeyourfuture.io/guides/deployment-netlify/) for your project.
```

Author: Thank you @illicitonion, I've removed the redundant deployment steps and streamlined the section to point to the official guide, avoiding duplication as suggested.
